src/corefx/System.Globalization.Native/CMakeLists.txt

```diff
@@ -61,9 +70,17 @@ add_library(System.Globalization.Native
 # Disable the "lib" prefix.
 set_target_properties(System.Globalization.Native PROPERTIES PREFIX "")
 
-target_link_libraries(System.Globalization.Native
-  ${ICUUC}
-  ${ICUI18N}
-)
+if(NOT CLR_CMAKE_PLATFORM_DARWIN)
+    target_link_libraries(System.Globalization.Native
+        ${ICUUC}
+        ${ICUI18N}
+    )
+else()
+    target_link_libraries(System.Globalization.Native
+        ${ICUCORE}
+    )
+
+    add_definitions(-DU_DISABLE_RENAMING=1)
```

Comment:
I will need to look into this a little bit more, I have refactored the CMakeLists.txt a bit so that on Darwin we don't probe headers for UDAT_STANDALONE_SHORTER_WEEKDAYS, not sure if that prevents me from doing this or not.  I'll investigate and do a follow-up PR if needed.